### DISCUSSION TITLE
Multiple argo controllers - how to point which one to use

### DISCUSSION THREAD

**Discussion by marioosh11** [argoproj/argo-rollouts#2389] [Q&A] [answered]
Hi Argo Lions
In our cluster we have dedicated namespace ago-rollouts where landed argo controller but we have limited permission to this namespace for example to use secret for notification.
We have a plan to install controller (only controller without CRDs) in own namespace.
Is there any field in spec in Rollout or any other way to use controller from given namespace?

BR
Mario

**Comment by zachaller** [comment, marked as answer]
We have this as in install methode https://github.com/argoproj/argo-rollouts/tree/master/manifests/namespace-install there are some issues with it that I bring up here https://github.com/argoproj/argo-rollouts/issues/2359

**Reply by zachaller** [reply]
There is also this https://github.com/argoproj/argo-rollouts/pull/342

`Adds a new flag to the controller called --instance-id. This flag indicates that the controller should only operate on Argo Rollout objects that have that label argo-rollouts.argoproj.io/controller-instance-id with the same value passed into the flag. If the flag is not listed, the controller operates on any Argo Rollout object that does not have the argo-rollouts.argoproj.io/controller-instance-id label.`

**Comment by marioosh11** [comment]
Hi @zachaller 
Thank you.
Our scope for now is only argo-rollout. We are installing argo-rollout via helm chart https://github.com/argoproj/argo-helm/blob/main/charts/argo-rollouts/ 
Would that mean that to have this feature we can inject --instance-id flag with value as extraEnv in controller yaml https://github.com/argoproj/argo-helm/blob/main/charts/argo-rollouts/templates/controller/deployment.yaml

Do you have any plan to parametrize it to have in helm values?

BR
Mario

**Reply by zachaller** [reply]
The helm chart is actually community maintained not by rollouts maintainers, it dose look like you could use this though https://github.com/argoproj/argo-helm/blob/main/charts/argo-rollouts/templates/controller/deployment.yaml#L43

**Reply by marioosh11** [reply]
Ok cool.
Thank you
